### PR TITLE
Update free surface cookbooks

### PR DESCRIPTION
**Pull request by gassmoeller**
Changes in this PR:
- rename the cookbooks free-surface and free-surface-with-crust to avoid dashes in favor of underscores (#3255)
- update the free_surface and free_surface_with_crust cookbooks to modern parameters
- bring the cookbooks into the new directory structure (#3422)
- improve some wording and the according figures
- fix a bug in the manual makefile that caused the makefile to generate .prm.out files from files like `cookbooks/free_surface/output-free_surface/parameters.prm`

### CODE REVIEW

File: doc/Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ change-files = $(shell find modules/changes/ -type f -not -name '.*')
 aspect-doc-prms = $(shell find -L manual/cookbooks/ -path \*/doc/\* -name \*.prm 2>/dev/null)
 aspect-doc-prms-out = $(aspect-doc-prms:=.out)
 
-aspect-cookbook-prms = $(shell find -L manual/cookbooks/ -not -path \*/doc/\* -name \*.prm 2>/dev/null)
+aspect-cookbook-prms = $(shell find -L manual/cookbooks/ -not -path \*/doc/\* -not -path \*output-\* -name \*.prm 2>/dev/null)
```
Comment (tjhei) [inline doc/Makefile:25]: Can you remove the dash after "output"? At least mid_ocean_ridge.prm output will not match this.

**Comment by gassmoeller** [comment]
I will try to reorganize this so you can distinguish moving the file from rewording the text.

**Comment by gassmoeller** [comment]
I separated the changes into three commits. The first reorganizes the files, separates the .tex files from the manual and updates .prm files and figures. The second updates the text of the cookbook documentation .tex files. The third addresses your Makefile comment.

**Pull request merged**